hbase_storage/src/main/java/com/splicemachine/si/impl/RollBackTxnFilter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2012 - 2021 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.si.impl;
+
+import com.clearspring.analytics.util.Lists;
+import com.splicemachine.access.api.Durability;
+import com.splicemachine.si.api.data.OperationFactory;
+import com.splicemachine.si.api.txn.Txn;
+import com.splicemachine.si.api.txn.TxnSupplier;
+import com.splicemachine.si.api.txn.TxnView;
+import com.splicemachine.si.constants.SIConstants;
+import com.splicemachine.si.impl.driver.SIDriver;
+import com.splicemachine.storage.*;
+import com.splicemachine.utils.ByteSlice;
+import com.splicemachine.utils.SpliceLogUtils;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.filter.Filter;
+import org.apache.hadoop.hbase.filter.FilterBase;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.log4j.Logger;
+
+import java.util.List;
+
+public class RollBackTxnFilter extends FilterBase {
```

Comment:
Why are we inheriting from `org.apache.hadoop.hbase.filter.FilterBase` instead of `com.splicemachine.si.api.filter.TxnFilter`?

Reply:
It is an HBase filter

Reply:
TxnFilter does not work with a HBase Get.